src/server/reply_send.c

```diff
@@ -175,10 +187,26 @@ dis_reply_write(int sfds, struct batch_request *preq)
 {
 	int rc;
 	struct batch_reply *preply = &preq->rq_reply;
+#ifndef WIN32
+	struct sigaction act, oact;
+#endif
+	time_t  old_tcp_timeout;
+	old_tcp_timeout = pbs_tcp_timeout;
+	pbs_tcp_timeout = PBS_DIS_TCP_TIMEOUT_REPLY;
 
 	if (preq->isrpp) {
 		rc = encode_DIS_replyRPP(sfds, preq->rppcmd_msgid, preply);
 	} else {
+#ifndef WIN32
+		/* set alarm to interrupt poll() etc. while flushing out data */
+		sigemptyset(&act.sa_mask);
+		act.sa_flags = 0;
+		act.sa_handler = reply_alarm;
+		if (sigaction(SIGALRM, &act, &oact) == -1)
+			return (PBS_NET_RC_RETRY);
+		reply_timedout = 0;
```

Comment:
For the first "Server-client" communication, reply_timedout is already 0 here:
https://github.com/PBSPro/pbspro/pull/1470/files#diff-921235111589f40507adc6d130c4e262R92

For subsequent communication, if the network is slow and that causes the timeout (trigger alarm), then the global variable is resetting here : https://github.com/PBSPro/pbspro/pull/1470/files#diff-921235111589f40507adc6d130c4e262R394

Again before de-registering the alarm we again reset back "reply_timedout = 0", this will help for next communication:
https://github.com/PBSPro/pbspro/pull/1470/files#diff-e304e748154092986f37050a075e0e54R225

Reply:
I understand what you are saying. I find it odd that you initialize the global variable that a signal handler is about set right after registering the handler. This may not be a problem for all practical reasons but Ideally, the initialization should happen before the handler is registered. Imagine if the signal was USR1 and right after registering the handler your program gets the signal, now the handler will set the global to 1 but soon after that, it will be reset to 0 because the initialization is happening right after registering the handler.

Reply:
Thank you @arungrover for the details. Yes, the initialization should happen before registering the Signal.